Assets/Scripts/Models/Time/WorldTime.cs

```diff
@@ -0,0 +1,508 @@
+    {
+        get
+        {
+            return (int)Hours / 24;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets the day component.
+    /// </summary>
+    /// <value>The day component.</value>
+    public int Day
+    {
+        get
+        {
+            return (Days % DaysPerQuarter) + 1;
+        }
+
+        set
+        {
+            Seconds -= Day * SecondsPerDay;
+            Seconds += value * SecondsPerDay;
+        }
+    }
+
+    /// <summary>
+    /// Gets the quarters since Epoch date (Midnight Q1 Day 1 2999).
+    /// </summary>
+    /// <value>The quarters since Epoch date (Midnight Q1 Day 1 2999).</value>
+    public int Quarters
+    {
+        get
+        {
+            return (int)Days / DaysPerQuarter;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets the quarter component.
+    /// </summary>
+    /// <value>The quarter component.</value>
+    public int Quarter
+    {
+        get
+        {
+            return (Quarters % 4) + 1;
+        }
+
+        set
+        {
+            Seconds -= Quarter * SecondsPerQuarter;
+            Seconds += value * SecondsPerQuarter;
+        }
+    }
+
+    /// <summary>
+    /// Gets the years since Epoch date (Midnight Q1 Day 1 2999).
+    /// </summary>
+    /// <value>The years since Epoch date (Midnight Q1 Day 1 2999).</value>
+    public int Years
+    {
+        get
+        {
+            return (int)Quarters / 4;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets the year component.
+    /// </summary>
+    /// <value>The year component.</value>
+    public int Year
+    {
+        get
+        {
+            return Years + StartingYear;
+        }
+
+        set
+        {
+            Seconds -= Years * SecondsPerYear;
+            Seconds += (value - StartingYear) * SecondsPerYear;
+        }
+    }
+
+    /// <summary>
+    /// Returns a string that represents the WorldTime time and date (separated by linebreak).
+    /// </summary>
+    /// <returns>A string that represents the WorldTime time and date.</returns>
+    /// <filterpriority>2</filterpriority>
+    public override string ToString()
+    {
+        // TODO: Optimally, we should have WorldTime as an IFormattable, so the format can be more easily adjusted, and doesn't require
+        // separate TimeToString and DateToString methods.
+        // Note: overloading is used, rather than defaults so that this plays nicely with Lua, which can't see default parameter values properly.
+        return ToString(true, true);
+    }
+
+    /// <summary>
+    /// Returns a string that represents the WorldTime, separated by a linebreak if both time and date are present.
+    /// </summary>
+    /// <returns>The string representing the WorldTime.</returns>
+    /// <param name="time">If set to <c>true</c> include the time in the string.</param>
+    /// <param name="date">If set to <c>true</c> include the date in the string.</param>
+    public string ToString(bool time, bool date)
```

Comment:
Can you have localization as a format string??  Though that has problems in itself maybe localize parts of it?